compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticWarningCode.java

```diff
@@ -69,7 +69,9 @@ public enum DiagnosticWarningCode implements DiagnosticCode {
 
     INVALID_METADATA_ON_DUPLICATE_ENUM_MEMBER("BCE20405", "invalid.metadata.on.duplicate.enum.member"),
 
-    USAGE_OF_STRAND_ANNOTATION_WILL_BE_DEPRECATED("BCE20406", "strand.annotation.will.be.deprecated")
+    USAGE_OF_STRAND_ANNOTATION_WILL_BE_DEPRECATED("BCE20406", "strand.annotation.will.be.deprecated"),
+
+    BUILT_WITH_OLDER_SL_UPDATE_DISTRIBUTION("BCE20407", "built.wtih.older.sl.update.distribution")
```

Comment:
We maintain `ProjectDiagnosticErrorCode` class for Project API diagnostics. Lets move the diagnostic there.  All resolution diagnostics begin with `BCE50`.

https://github.com/ballerina-platform/ballerina-lang/blob/b43096b16218cd6c7f0158eb4582089690ddb09c/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ProjectDiagnosticErrorCode.java#L30-L37

The DiagnosticId for this warning will be `BCE5008`